datalad/tests/test_annexrepo.py

```diff
@@ -219,3 +234,57 @@ def test_AnnexRepo_annex_add_to_git(src, dst):
     assert_raises(IOError, ar.get_file_key, filename)
     ar.annex_add_to_git(filename)
     assert_in(filename, ar.get_indexed_files())
+
+
+@with_testrepos(flavors=local_flavors)
+@with_tempfile
+def test_AnnexRepo_web_remote(src, dst):
+
+    ar = AnnexRepo(dst, src)
+
+    testurl = 'http://datalad.org/pages/about.html'
+    testfile = 'datalad.org_pages_about.html'
```

Comment:
I still wonder how could we mock external annex... but for this test -- do we really need remote location?  there is ```@serve_path_via_http``` which is often used together with ```@with_tree``` so you could provide a proper http url coming from local box